cscs-checks/analytics/spark/spark_check.py

```diff
@@ -0,0 +1,44 @@
+import os
+import math
+
+import reframe.utility.sanity as sn
+from reframe.core.pipeline import RunOnlyRegressionTest
+from reframe.core.launchers import LauncherWrapper
+from reframe.core.launchers.registry import getlauncher
+
+
+class SparkAnalyticsCheck(RunOnlyRegressionTest):
+    def __init__(self, **kwargs):
+        super().__init__('spark_check',
+                         os.path.dirname(__file__), **kwargs)
+        self.descr = 'Simple calculation of pi with Spark'
+        self.valid_systems = ['daint:gpu', 'daint:mc']
+        self.valid_prog_environs = ['PrgEnv-cray']
+
+        self.modules = ['analytics']
+
+        self.executable = 'start_analytics -t "spark-submit spark_pi.py"'
+
+        pi_value = sn.extractsingle(r'Pi is roughly\s+(?P<pi>\S+)',
+                                    self.stdout, 'pi', float)
+        pi_reference = math.pi
+        pi_diff = sn.abs(pi_value - pi_reference)
+
+        self.sanity_patterns = sn.assert_lt(pi_diff, 0.01)
```

Comment:
If it fits in a single line, you may replace `pi_diff` with the `sn.abs()` expression directly.